### PR TITLE
[RHMAP 5796] Support new Environments model on FHC

### PR DESCRIPTION
**Pull request by paolobueno**
Remove `fhc admin environments alias` commands and update commands to utilize the human-readable `id`.

Also update test fixtures and http mocks to reflect the new api.

ping @pb82 @odra @wei-lee

### CODE REVIEW

**Comment by fh-automatron** [comment]
### Pull Request Quality Guidelines

As we continue to focus on quality, here are a few things to keep in mind:

* The code changes conform to the contributing & code style guides for this component (if they exist)
* A component owner should review your change (we've tagged these above)
* Automated testing has been added or updated for this change (e.g. UART, Unit/Acceptance tests)
* There should be an associated fhcap PR with this change - please provide a link to this
* If updating server component this PR should be deployed to a grid somewhere (via `fhcap-cli`) - please provide a link to this
* Ensure JIRA is up to date before & after merge (e.g. deployment notes, tester notes, release notes, fix version)
* Any SonarQube issues should be addressed where applicable (use your own judgement)

Please ping a component owner (see `fh contrib`) when your builds are complete, changes are deployed and you're ready for a review.

In order to update fhcap with your component change please execute:
`fhcap component update fh-fhc 2.6.0 266`

File: lib/cmd/fh3/admin/environments/create.js [hunk 1]
```diff
@@ -2,41 +2,40 @@ var log = require("../../../../utils/log");
...
     }
     var request = {
-      _id: params.id,
+      id: params.id,
       label: params.label,
-      targets: targets,
+      target: params.target,
       autoDeployOnCreate: params.autoDeployOnCreate,
       autoDeployOnUpdate: params.autoDeployOnUpdate,
       noAppend: params.noAppend
```
Comment (wei-lee) [inline lib/cmd/fh3/admin/environments/create.js]: I think we can remove this flag now (as well as the command flag). It is not used anymore.

File: lib/cmd/fh3/admin/environments/create.js [hunk 2]
```diff
@@ -2,41 +2,40 @@ var log = require("../../../../utils/log");
...
     };
 
-    if(params.noAppend){
+    if(params.noAppend) {
```
Comment (wei-lee) [inline lib/cmd/fh3/admin/environments/create.js]: this can be removed too

**Comment by wei-lee** [comment]
LGTM

File: lib/cmd/fh3/admin/environments/create.js [hunk 3]
```diff
@@ -1,44 +1,35 @@
...
-  'demand' : ['id', 'label', 'targets'],
+  'demand' : ['id', 'label', 'target'],
   'alias' : {},
   'describe' : {
     'id' : 'Unique identifier for your environment',
     'label' : 'A label describing your environment',
-    'targets' : 'Comma separated list of MBaaS Target hostnames',
+    'target' : 'unique identifier for an existing MBaaS Target',
     'autoDeployOnCreate' : 'option used to setup automatic deployment to this environment when creating new project/app.',
     'autoDeployOnUpdate' : 'option used to setup automatic deployment to this environment when making source code changes within app.',
```
Comment (fh-qube) [inline lib/cmd/fh3/admin/environments/create.js]: MAJOR Remove this trailing comma. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingComma)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by paolobueno** [comment]
@wei-lee Thanks for review. I've removed the `noAppend` flag, makes a lot of sense since it wasn't present on `update` either and nowhere else.

Merging to feature branch...

**Comment by ciaran-byrne** [comment]
The noAppend flag needs to be put back: https://issues.jboss.org/browse/RHMAP-3295

**Comment by paolobueno** [comment]
Reenabling on #203

**Comment by vchepeli** [comment]
@ciaran-byrne yep but should it be noAppend or no-append? Cause @jmernin once told that we should use some specific style for opts

**Comment by vchepeli** [comment]
@paolobueno, please next time add RHMAP-XXXX prefix to all commit messages, then it's easy to find what task this commits belongs to and we comment there

**Comment by wei-lee** [comment]
@ciaran-byrne Sorry, I just saw the comment. I don't really think we need that flag anymore - we have removed the "environments.available" property, and by default newly created mbaas target or environment will only be available at the domain level.  Even we add back to fhc, the API endpoint will not  use it.

**Comment by jessesarn** [comment]
@wei-lee So how do we control the order of the environments list if we can no longer use a property to force it?

**Comment by wei-lee** [comment]
@jessesarn The customers (or ops team) can just use the admin view in the studio to change the order of environments. TBH, I think customers can do it by themselves. ops team do not need to do it anymore.

**Pull request merged**